src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -156,14 +222,30 @@ protected KualiDecimal calculateTotalAmountForLines(List<ConcurStandardAccountin
     protected Optional<OriginEntryFull> buildOriginEntryForCashOffset(
             OriginEntryFull cashEntry, List<ConcurStandardAccountingExtractDetailLine> cashLines) {
         KualiDecimal cashAmount = getSignedAmountFromOriginEntry(cashEntry);
+        KualiDecimal cashAdvanceAmount = calculateTotalAmountForCashAdvanceLinesReferencedByRegularLines(cashLines);
+        KualiDecimal cashAmountToOffset = cashAmount.add(cashAdvanceAmount);
+        if (cashAmountToOffset.isZero()) {
+            return Optional.empty();
+        }
         
-        OriginEntryFull offsetEntry = buildOffsetOriginEntry(cashEntry, cashAmount);
+        OriginEntryFull offsetEntry = buildOffsetOriginEntry(cashEntry, cashAmountToOffset);
         offsetEntry.setFinancialObjectCode(collectorHelper.getCashOffsetObjectCode());
         offsetEntry.setFinancialSubObjectCode(collectorHelper.getDashOnlyPropertyValue(KFSPropertyConstants.SUB_OBJECT_CODE));
         
         return Optional.of(offsetEntry);
     }
 
+    protected KualiDecimal calculateTotalAmountForCashAdvanceLinesReferencedByRegularLines(
+            List<ConcurStandardAccountingExtractDetailLine> regularDetailLines) {
+        return regularDetailLines.stream()

```

Comment:
Can this statement be refactored for clarity?  Maybe creation of some well named variables would help.